### PR TITLE
fix: use autocomplete field for selecting job

### PR DESCRIPTION
**Pull request by cdeery**
The Job chooser in the B2CJobAllowlist admin is not usable in production because there are too many jobs for a usable popup menu.
This fix configures the admin to use an autocomplete menu that enables searching the list of jobs.

**Merge checklist:**
- [x] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `make upgrade && make requirements` have been run to regenerate requirements
- [x] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
- [x] [Version](https://github.com/openedx/taxonomy-connector/blob/master/taxonomy/__init__.py) bumped
- [x] [Changelog](https://github.com/openedx/taxonomy-connector/blob/master/CHANGELOG.rst) record added

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/taxonomy-connector/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/openedx/taxonomy-connector/actions), verify version has been pushed to [PyPI](https://pypi.org/project/taxonomy-connector/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [course-discovery](https://github.com/openedx/course-discovery) to upgrade dependencies (including taxonomy-connector)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in course-discovery will look for the latest version in PyPi.

### CODE REVIEW

File: taxonomy/admin.py [hunk 1]
```diff
@@ -220,6 +220,7 @@ class B2CJobAllowListAdmin(admin.ModelAdmin):
 
     list_display = ('id', 'external_id', 'job_name',)
     search_fields = ('job__name', 'job__external_id',)
+    autocomplete_fields = ('job',)
```
Comment (cdeery) [inline taxonomy/admin.py:223]: This is the only actual code change - this line was added.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -13,6 +13,10 @@ Change Log
 
 Unreleased
 
+[1.42.0] - 2023-07-06
```
Comment (muhammad-ammar) [inline CHANGELOG.rst:16]: we also need to update the version here. `1.42.0` -> `1.42.1`

Comment (cdeery) [reply CHANGELOG.rst]: Thank you! Copy and paste error, my apologies. Corrected now

**Review by muhammad-ammar — APPROVED** [review]
LGTM

**Pull request merged**